### ISSUE TITLE
ejabberd should support namespace prefixes

### ISSUE THREAD

**Issue by sezuan** [processone/ejabberd#814]
see also
- https://github.com/edhelas/moxl/issues/10
- https://support.process-one.net/browse/EJAB-1661

**Comment by horazont** [comment]
Reproducer for STARTTLS:

works: (server replies with proceed)
```xml
<?xml version='1.0'?>
<stream:stream
          to='jabber.ccc.de'
          version='1.0'
          xml:lang='en'
          xmlns='jabber:client'
          xmlns:stream='http://etherx.jabber.org/streams'>
<starttls xmlns='urn:ietf:params:xml:ns:xmpp-tls'/>
```

fails: (creates stream error, policy violation indicating that STARTTLS is required)
```xml
<?xml version='1.0'?>
<stream:stream
          to='jabber.ccc.de'
          version='1.0'
          xml:lang='en'
          xmlns='jabber:client'
          xmlns:stream='http://etherx.jabber.org/streams'>
<ns0:starttls xmlns:ns0='urn:ietf:params:xml:ns:xmpp-tls'/>
```

**Comment by prefiks** [comment]
This should be fixed by commit a25471470a9883fd64f3bbe20b5f1b1f4253137f
